src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -559,5 +559,90 @@ static void M(ParameterSyntax parameter)
     }
 }");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task InlineTypeCheck3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        [|string|] x;
+        if ((x = o as string) != null)
```

Comment:
We'll need to figure out if we need a new option here.  Or if we need to update the option we show to the user here.  I think it's ok to have a single option, but we likely need to call out these cases in the preview.  What do you think @Pilchie ?